go/tasks/plugins/k8s/kfoperators/common/common_operator.go

```diff
@@ -0,0 +1,125 @@
+package common
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	commonOp "github.com/kubeflow/tf-operator/pkg/apis/common/v1"
+	logUtils "github.com/lyft/flyteidl/clients/go/coreutils/logs"
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+	flyteerr "github.com/lyft/flyteplugins/go/tasks/errors"
+	"github.com/lyft/flyteplugins/go/tasks/logs"
+	pluginsCore "github.com/lyft/flyteplugins/go/tasks/pluginmachinery/core"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	TensorflowTaskType = "tensorflow"
+	PytorchTaskType    = "pytorch"
+)
+
+func ExtractCurrentCondition(jobConditions []commonOp.JobCondition) (commonOp.JobCondition, error) {
+	sort.Slice(jobConditions[:], func(i, j int) bool {
+		return jobConditions[i].LastTransitionTime.Time.After(jobConditions[j].LastTransitionTime.Time)
+	})
+
+	for _, jc := range jobConditions {
+		if jc.Status == v1.ConditionTrue {
+			return jc, nil
+		}
+	}
+
+	return commonOp.JobCondition{}, fmt.Errorf("found no current condition. Conditions: %+v", jobConditions)
+}
+
+func GetPhaseInfo(currentCondition commonOp.JobCondition, occurredAt time.Time,
+	taskPhaseInfo pluginsCore.TaskInfo) (pluginsCore.PhaseInfo, error) {
+	switch currentCondition.Type {
+	case commonOp.JobCreated:
+		return pluginsCore.PhaseInfoQueued(occurredAt, pluginsCore.DefaultPhaseVersion, "JobCreated"), nil
+	case commonOp.JobRunning:
+		return pluginsCore.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion, &taskPhaseInfo), nil
+	case commonOp.JobSucceeded:
+		return pluginsCore.PhaseInfoSuccess(&taskPhaseInfo), nil
+	case commonOp.JobFailed:
+		details := fmt.Sprintf("Job failed:\n\t%v - %v", currentCondition.Reason, currentCondition.Message)
+		return pluginsCore.PhaseInfoRetryableFailure(flyteerr.DownstreamSystemError, details, &taskPhaseInfo), nil
+	case commonOp.JobRestarting:
```

Comment:
Restarting means it's already going to get retried on its own, right? returning RetryableFailure would actually make flyte kill the existing instance and retry...
If my understanding is correct, this should return Running

Reply:
This snippet of code is copied from https://github.com/flyteorg/flyteplugins/blob/8c66bddd184ea1d605e74ea79177df7a4fdf124c/go/tasks/plugins/k8s/kfoperators/pytorch/pytorch.go#L134-L149

I think if the job was failed maybe there is something wrong like the docker image can not pull, so we should add a job failed message to the log.
btw, if I change to `pluginsCore.PhaseInfoRunnin`, the test will be broken.
If we make sure to change to `running`, I will change test sets as well.

Reply:
@pingsutw do you know of cases in which Operator status will be restarting? I can maybe look at the code. 
But @EngHabu I think I agree that Pytorch and TFoperator should be the same as Kf has common status across both.

Reply:
I understand @pingsutw this is copied code. I'm not sure this particular case was tested before to be honest.
[This is the snippet ](https://github.com/kubeflow/tf-operator/blob/4f6906951804ff4530ffe2cb9567b19f1eeb760d/pkg/controller.v1/tensorflow/pod.go#L135-L157) of the code that sets the status to restarting.
It sounds like from Flyte perspective, it'll keep running... 

So yes, we will also need to update the test.

Are you referring to [this](https://github.com/kubeflow/common/blob/master/pkg/apis/common/v1/types.go#L118-L122) for the common code?

I agree we should replicate what kf operators common pattern has...

Reply:
I've updated to `pluginsCore.PhaseInfoRunnin`